File: gaia-sdk-javascript/src/graphql/request/type/ConnectNodeRemovedImpulse.ts

import {EdgeKeyOne} from "./EdgeKeyOne";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result removing a node connection
 */
export class ConnectNodeRemovedImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "ConnectNodeRemovedImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * edges removed by this request
     */
    public removedEdges = (config: (_:EdgeKeyOne) => void) => this.push((registry) => {
        const entity = new EdgeKeyOne();
        config(entity);
        return "removedEdges { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/ConnectNodeRemovedImpulse.py

from gaia_sdk.graphql.request.type.EdgeKeyOne import EdgeKeyOne

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class ConnectNodeRemovedImpulse(list):
    """
    Impulse which indicates the result removing a node connection
    """

    def id(self):
        self.append(lambda x: "id")

    """
    edges removed by this request
    """
    def removed_edges(self, config: Callable[['EdgeKeyOne'], None]):
        def callback(registry: VariableRegistry):
            entity = EdgeKeyOne()
            config(entity)
            return "removed_edges {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
